src/components/Missions.js

```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { joinMission, leaveMission } from '../redux/missions/missionsSlice';
+import './Missions.css';
+
+const Missions = () => {
+  const dispatch = useDispatch();
+  const missions = useSelector((state) => state.missions.missions);
+  const status = useSelector((state) => state.missions.status);
+  const error = useSelector((state) => state.missions.error);
+
+  const handleJoinMission = (missionId) => {
+    dispatch(joinMission(missionId));
+  };
+
+  const handleLeaveMission = (missionId) => {
+    dispatch(leaveMission(missionId));
+  };
+
+  if (status === 'loading') {
+    return <div>Loading...</div>;
+  }
+
+  if (status === 'failed') {
+    return (
+      <div>
+        Error:
+        {error}
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <h2 className="h2-missions">Missions</h2>
+      <table>
+        <thead>
+          <tr>
+            <th>Mission Name</th>
+            <th>Description</th>
+            <th>Status</th>
+            <th>Join/Leave</th>
+          </tr>
+        </thead>
+        <tbody>
+          {missions.map((mission) => {
+            const isMember = mission.reserved;
+            const handleButtonClick = isMember ? handleLeaveMission : handleJoinMission;
+            const buttonDisabled = isMember ? false : mission.reserved;
+
+            return (
+              <tr key={mission.mission_id}>
+                <td>{mission.mission_name}</td>
+                <td>{mission.description}</td>
+                <td className="status">
+                  <div className={isMember ? 'member' : 'not-member'}>
+                    {isMember ? 'Active Member' : 'NOT A MEMBER'}
+                  </div>
+                </td>
+                <td>
+                  <button type="button" className={isMember ? 'leave-button' : 'join-button'} onClick={() => handleButtonClick(mission.mission_id)} disabled={buttonDisabled}>
+                    {isMember ? 'Leave Mission' : 'Join Mission'}
+                  </button>
+                </td>
```

Comment:
- [ Required ] I think it would be better to use conditional components rendering. It's considered good practice not to handle app logic of any kind in a React component, a good React component should only be used for display purposes. I would like to suggest that you refactor this code to use conditional components as shown below.

```JavaScript
  return (
    <div>
      <h2 className="h2-missions">Missions</h2>
      <table>
        <thead>
          <tr>
            <th>Mission Name</th>
            <th>Description</th>
            <th>Status</th>
            <th>Join/Leave</th>
          </tr>
        </thead>
        <tbody>
          {missions.map((mission) => (
            <tr key={mission.mission_id}>
              <td>{mission.mission_name}</td>
              <td>{mission.description}</td>
              <td className="status">
                {(mission.reserved && <div className="member">Active Member</div>)}
                {(!mission.reserved && <div className="member">NOT A MEMBER</div>)}
              </td>
              <td>
                {(mission.reserved && (
                  <button
                    type="button"
                    className="leave-button disabled"
                    onClick={() => handleLeaveMission(mission.mission_id)}
                  >
                    Leave Mission
                  </button>
                ))}

                {(!mission.reserved && (
                  <button
                    type="button"
                    className="join-button"
                    onClick={() => handleJoinMission(mission.mission_id)}
                  >
                    Join Mission
                  </button>
                ))}
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
```
## Project requirements
![Screenshot from 2023-04-14 23-37-38](https://user-images.githubusercontent.com/16392046/232150490-8875c1b8-153a-4fc0-9da1-455fb45949f3.png)